dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/apphub/DefaultAppHubService.java

```diff
@@ -104,6 +105,18 @@ public List<WebApp> getAppHub()
         return Arrays.asList( apps );
     }
 
+    @Override
+    public AppVersion getWebAppVersion( String versionId )
+    {
+        validateUuid( versionId );
```

Comment:
Doesn't seem like this is enough to fix the "security vulnerability" reported. Do we need a regex-check, or how can we resolve it?

Reply:
Is the exception from the validation actually thrown here?

See my comment about the validation util method; You could rewrite this to a simple if-else, that would completely let you ignore the real logic if the validation fails.

Reply:
Well, validateUuid will throw if validation fails. So then it would propagate here and throw as well.